poetry.lock

```diff
@@ -497,6 +508,36 @@ urllib3 = ">=1.21.1,<1.27"
 socks = ["PySocks (>=1.5.6,!=1.5.7)", "win-inet-pton"]
 use_chardet_on_py3 = ["chardet (>=3.0.2,<5)"]
 
+[[package]]
+name = "sentry-sdk"
+version = "1.5.5"
+description = "Python client for Sentry (https://sentry.io)"
+category = "main"
+optional = false
```

Comment:
This one alarmed me. The intent is to make `sentry-sdk` optional at runtime but required at dev-time.

But I think this is actually fine as it is. A fresh `--no-dev` installation created as follows didn't include sentry.

```shell
poetry shell
# nuke poetry's venv
pip freeze | cut -d " " -f 1 | xargs pip uninstall -y
poetry install --no-dev
pip list | grep 'sentry' # no results
```

A fresh install with dev dependencies _did_ include sentry:

```shell
pip freeze | cut -d " " -f 1 | xargs pip uninstall -y
poetry install
...
  • Installing sentry-sdk (1.5.5)
...

And for completeness I checked what happened when installing from a wheel. That didn't install sentry.

```shell
poetry build
cd /tmp
python -m venv env
source env/bin/activate
pip install ~/workspace/sydent/dist/matrix_sydent-2.5.1-py3-none-any.whl
pip list | grep sentry  # no output
```